app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
```

Comment:
[OPTIONAL] I think your code will be more concise if you had just done this  author.increment!(:posts_counter). You can also include a after_save callback and make the method private this is a good practice.

Reply:
Done